### PR TITLE
Update/add Warning, Info, Check, Cross, TableConfig icons

### PR DESCRIPTION
**Pull request by jacklenehan**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/components/Icon/InfoIcon/InfoIcon.jsx [hunk 1]
```diff
@@ -15,6 +15,12 @@ const InfoIcon = React.createClass({
 		...Icon.propTypes,
 	},
 
+	getDefaultProps() {
+		return {
+			viewBox: '-23.5 0.5 16 16',
```
Comment (jondlm) [inline src/components/Icon/InfoIcon/InfoIcon.jsx]: We can leave this out and modify the rects a little bit:

```
<rect x="7" y="3" width="2" height="2" data-reactid=".0.1.0.6.$Default.2.0.0"></rect>
<rect x="7" y="6" width="2" height="7" data-reactid=".0.1.0.6.$Default.2.0.1"></rect>
```

File: src/components/Icon/WarningIcon/WarningIcon.jsx [hunk 2]
```diff
@@ -32,9 +32,9 @@ const WarningIcon = React.createClass({
 				{...passThroughs}
 				className={boundClassNames('&', className)}
 			>
-				<path className={boundClassNames('&-triangle')} d='M9,1C8.2,1,7.4,1.6,7.1,2.3C5.9,4.3,1,12.9,0.5,13.7C-0.6,15.5,0.3,17,2.4,17h13.2 c2.1,0,3-1.5,1.9-3.3C14.9,9.2,13.7,6.9,11,2.3C10.6,1.5,9.9,1,9,1L9,1z'/>
-				<rect x='8.1' y='5.4' fill='#ffffff' width='1.8' height='6.3'/>
-				<rect x='8.1' y='12.6' fill='#ffffff' width='1.8' height='1.8'/>
+				<path className={boundClassNames('&-triangle')} d='M-14.489,0.5c-0.632,0-1.264,0.457-1.743,1.369c-1.026,1.952-5.404,10.463-5.849,11.312  c-0.959,1.822-0.191,3.318,1.705,3.318c1.062,0,8.613,0,11.753,0c1.896,0,2.664-1.493,1.707-3.318  c-2.375-4.525-3.457-6.789-5.829-11.312C-13.227,0.958-13.859,0.5-14.489,0.5L-14.489,0.5z'/>
```
Comment (jondlm) [inline src/components/Icon/WarningIcon/WarningIcon.jsx]: Here's cleaner svg code for a 16x16 viewbox:

```
<path className={boundClassNames('&-triangle')} d='M8.011,0 C7.379,0 6.747,0.457 6.268,1.369 C5.242,3.321 0.864,11.832 0.419,12.681 C-0.54,14.503 0.228,15.999 2.124,15.999 C3.186,15.999 10.737,15.999 13.877,15.999 C15.773,15.999 16.541,14.506 15.584,12.681 C13.209,8.156 12.127,5.892 9.755,1.369 C9.273,0.458 8.641,0 8.011,0 L8.011,0 z'/>

<rect x='7' y='4' fill='#FFFFFF' width='2' height='7'></rect>
<rect x='7' y='12' fill='#FFFFFF' width='2' height='2'></rect>
```

File: src/components/Icon/TableConfigIcon/TableConfigIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			className,
+			...passThroughs
+		} = this.props;
+
+		return (
+			<Icon
+				{...passThroughs}
+				className={boundClassNames('&', className)}
+			>
+			<rect className={boundClassNames('&-background')} x='1' y='4' width='13' height='8'/>
```
Comment (jondlm) [inline src/components/Icon/TableConfigIcon/TableConfigIcon.jsx]: Indentation looks funky here

**Comment by ogupte** [comment]
I think the icon name should be more about what it displays than what it's used for: TableCog might be a more appropriate name for this icon. This is more like how other icon libs/sets name their icons.

**Comment by jacklenehan** [comment]
Makes sense, changing. Also fixing indentation.

**Comment by jacklenehan** [comment]
Fixes #240, #242, #243

**Pull request merged**